contracts/WhitelistedRecipientsRegistry.sol

```diff
@@ -0,0 +1,142 @@
+    ///     granted with role ADD_WHITELISTED_RECEPIENT_ROLE
+    /// @param _removeWhitelistedRecipientRoleHolders List of addresses which will
+    ///     be granted with role REMOVE_WHITELISTED_RECEPIENT_ROLE
+    constructor(
+        address _admin,
+        address[] memory _addWhitelistedRecipientRoleHolders,
+        address[] memory _removeWhitelistedRecipientRoleHolders,
+        address[] memory _setLimitParametersRoleHolders,
+        EasyTrack _easy_track
+    ) LimitsChecker(_easy_track, _setLimitParametersRoleHolders) {
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        for (uint256 i = 0; i < _addWhitelistedRecipientRoleHolders.length; i++) {
+            _setupRole(ADD_WHITELISTED_RECEPIENT_ROLE, _addWhitelistedRecipientRoleHolders[i]);
+        }
+        for (uint256 i = 0; i < _removeWhitelistedRecipientRoleHolders.length; i++) {
+            _setupRole(
+                REMOVE_WHITELISTED_RECEPIENT_ROLE,
+                _removeWhitelistedRecipientRoleHolders[i]
+            );
+        }
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Adds address to list of allowed addresses for payouts
+    function addWhitelistedRecipient(address _whitelistedRecipient, string memory _title)
+        external
+        onlyRole(ADD_WHITELISTED_RECEPIENT_ROLE)
+    {
+        require(
+            whitelistedRecipientIndices[_whitelistedRecipient] == 0,
+            ERROR_WHITELISTED_RECEPIENT_ALREADY_ADDED
+        );
+
+        whitelistedRecipients.push(_whitelistedRecipient);
+        whitelistedRecipientIndices[_whitelistedRecipient] = whitelistedRecipients.length;
+        emit WhitelistedRecipientAdded(_whitelistedRecipient, _title);
+    }
+
+    /// @notice Removes address from list of allowed addresses for payouts
+    /// @dev To delete an allowed address from the whitelistedRecipients array in O(1),
+    /// we swap the element to delete with the last one in the array,
+    /// and then remove the last element (sometimes called as 'swap and pop').
+    function removeWhitelistedRecipient(address _whitelistedRecipient)
+        external
+        onlyRole(REMOVE_WHITELISTED_RECEPIENT_ROLE)
+    {
+        uint256 index = _getWhitelistedRecipientIndex(_whitelistedRecipient);
+        uint256 lastIndex = whitelistedRecipients.length - 1;
+
+        if (index != lastIndex) {
+            address lastWhitelistedRecipient = whitelistedRecipients[lastIndex];
+            whitelistedRecipients[index] = lastWhitelistedRecipient;
+            whitelistedRecipientIndices[lastWhitelistedRecipient] = index + 1;
+        }
+
+        whitelistedRecipients.pop();
+        delete whitelistedRecipientIndices[_whitelistedRecipient];
+        emit WhitelistedRecipientRemoved(_whitelistedRecipient);
+    }
+
+    /// @notice Returns if passed address are listed as whitelisted recipient in the registry
+    function isWhitelistedRecipient(address _maybeWhitelistedRecipient)
```

Comment:
I'd prefer to go with just `_address` instead of `_maybeWhitelistedRecipient` as a more plain clear name